docs/explanations/rendering.md

```diff
@@ -0,0 +1,344 @@
+take.
+
+In #845 the array access leads to loading data from disk or over the network. In
+#1320 the array access leads to a machine learning calculation with Torch. In
+#1300 the problem is different. There the data is already entirely in memory,
+but it's not chunked. Today that means we transfer 100's of MB to the card in
+one shot and this blocks the UI. We can't allow huge monolithic arrays of data
+in napari. Everything must be broken down into reasonably sized chunks.
+
+# Requirements
+
+To met our goal of never blocking the UI we need to satisfy two requirements:
+
+1. Always break data into chunks. The exact maximum chunk size is TBD.
+2. Never call `asarray` on user data from the GUI thread since we don't know
+   what it will do or how long it will take.
+
+# Render Algorithm
+
+The renderer will intersect the current view with the dataset to determine the
+working set. The working set is the set of chunks that we need to draw to fully
+render that specific view. The renderer will step through every chunk in the
+working set and do one of three things:
+
+| Case                         | Action                                      |
+| ---------------------------- | ------------------------------------------- |
+| Chunk is in VRAM             | Render the chunk                            |
+| Chunk is in RAM but not VRAM | Transfer the chunk to VRAM if there is time |
+| Chunk is not in RAM          | Ask the `ChunkManager` to load the chunk    |
+
+If a chunk cannot be drawn a placeholder will be drawn instead. What we draw as
+a placeholder is TBD and it may vary. In some cases we might be able to draw a
+lower resolution version of the data, which can be refined later as more data is
+available. In the worst case we might have to draw a blank grid or a loading
+animation.
+
+# Chunks
+
+**Chunks** is a deliberately vague term. *A chunk is data used to render a
+portion of the scene*. Without chunks we have only two choices: render nothing
+or render the entire scene. With chunks we can partially render the scene using
+whatever chunks are currently available, and we can progressively add more and
+more chunks as more data is loaded.
+
+Progressive rendering is valuable because the user can often navigate or make
+other decisions with partially loaded data, so the user can work faster. Also
+progressive rendering just feels more pleasant for the user. Progressive rending
+makes the internal state of the application visible which is often leads to a
+better user experience.
+
+## Chunked File Formats
+
+The most common types of chunks are blocks of contiguous memory inside a chunked
+file format like **Zarr** and exposed by an API like **Dask**. If an
+image is stored without chunks then reading a 2D rectangle would require
+hundreds of small read operations from all over the file. With chunks you can
+read a rectangular region with a single read operation.
```

Comment:
i feel like this sentence could be clearer or more precise (but I don't have a suggestion)

Reply:
I used more sentences it now says: "If an image is stored without chunks then reading any given small 2D region of the image requires many different read operations. The bytes for that region are spread throughout the file, intermingled with the bytes from other regions. With chunking you can read a rectangular region with a single read operation."

Chunking is basic stuff to many developers. But I think it's a kind of detail that a someone new to this stuff might not get at first.